### DISCUSSION TITLE
Collecting New Features for Release 2.7

### DISCUSSION THREAD

**Discussion by STRRL** [chaos-mesh/chaos-mesh#4094] [Ideas]
Hello everyone!

As the maintainer of our open-source project, I am excited to announce the next release is version 2.7. To make this release the best it can be and serve our community's needs, I'm reaching out to ask for your help in collecting new features.

If you have any ideas or suggestions for features that you would like to see in the upcoming release, please share them in this thread. We want to hear from you and ensure that the new release addresses the needs of our users and contributors as much as possible.

Here are some guidelines to follow while sharing your suggestions:

- Give a detailed explanation of the new feature you want to be included in the release.
- Explain the use cases and benefits of the feature.
- If possible, provide your implementation suggestions.
- Any suggestions that are aligned with our project's goals and values, and follow our code of conduct, will be considered.

Please contribute your ideas so that we can make version 2.7 a great success! Thank you in advance for your help and support.

Thanks!

**Comment by g1eny0ung** [comment]
Here are my suggestions for changes to the dashboard UI:

- [ ] https://github.com/chaos-mesh/chaos-mesh/issues/4096
- [ ] https://github.com/chaos-mesh/chaos-mesh/issues/4098

**Comment by see-quick** [comment]
Hi, what about adding a Kubernetes experiment which triggers to Kubernetes node to crash/failure/restart/stop? 

1. Explain the use cases and benefits of the feature:

As a user of chaos engineering, I want to operate on a node level and create a hypothesis such as "What happens to my application when one of my Kubernetes nodes crashed/failure/restart?", What happens to my application if there is no enough memory or CPU worker node left on worker node? It would be good to see the results of such experiments I think.

2. If possible, provide your implementation suggestions.

I am not familiar with you codebase but first we should decide if this idea make a sense. If so we could proceed with the design...

**Comment by jrhunger** [comment]
Generally, would be good to consider the use case of using the console primarily to design experiments that will be exported and run in other ways (e.g. in pipelines).
- Add a more direct way to export workflow YAML from the designer.  You can currently press submit and it's in the preview, on the intermediate page, but would be nice to have an Export Workflow button beside the Import Workflow button on the designer.
- That would also help to locally save a WIP workflow to Import later and work on some more
- Similarly in the single Experiment page it would be nice if the By YAML tab populates from what is chosen on the New Experiment tab, and vice-versa.  So you could load a YAML, click to the designer and change some parameters and submit.  Or vice-versa design the experiment through the designer and switch to the YAML tab and copy/paste to save in your pipeline test.
- Someone with Viewer role should be able to use it in this way as well (i think this may be true now, just getting an error if actually submitting?)

**Comment by jrhunger** [comment]
Another idea for the UI is the some way to hide buttons that are not relevant.  For example at least two if not three of aws/azure/gcp are irrelevant for every cluster and could be hidden.  And in a shared cluster, a namespace user probably should only have access to experiments that are limited to the namespace.  Maybe only show buttons for which the authenticated user/token has privileges to the underlying CRD?

**Comment by george-tj** [comment]
s there a plan to support HTTPS fault injection
In most real business scenarios, HTTPS is used, and when using the Chaos mesh platform for injection, they do not want to modify anything

**Comment by PRO-INDIA** [comment]
Currently an experiment created runs only once and to rerun we need to recreated the experiment or upload the yaml file again. This keeps on accumulating the experiments till we archive it.

It would be nice to have feature where we can create /configure an experiment and rerun the same whenever required. This would save a lot of time and effort.
